runtime/service.go

```diff
@@ -1182,43 +1183,39 @@ func (s *service) shutdown(
 ) error {
 	shutdownCtx, cancel := context.WithTimeout(requestCtx, timeout)
 	defer cancel()
-	go func() {
-		// Once the shutdown procedure is done, the shim needs to shutdown too.
-		// This also ensures that if the VMM is still alive, it will receive a
-		// SIGKILL via exec.CommandContext
-		<-shutdownCtx.Done()
-		s.shimCancel()
-	}()
 
 	s.logger.Info("stopping the VM")
 
-	// Try to tell agent to exit, causing the VM to begin a reboot. If that
-	// fails, try to forcibly stop the VMM. If that too fails, just cancel
-	// the shutdownCtx to fast-path to the VMM getting SIGKILL.
-	_, shutdownErr := s.agentClient.Shutdown(shutdownCtx, req)
-	if shutdownErr != nil {
-		s.logger.WithError(shutdownErr).Error("failed to shutdown VM agent")
-		stopVMMErr := s.machine.StopVMM()
-		if stopVMMErr != nil {
-			s.logger.WithError(stopVMMErr).Error("failed to forcibly stop VMM")
-			cancel()
+	go func() {
+		s.logger.Debug("shutdown through VM agent")
+		_, err := s.agentClient.Shutdown(shutdownCtx, req)
+		if err != nil {
+			s.logger.WithError(err).Error("failed to send shutdown")
+		} else {
+			time.Sleep(timeout)
 		}
-	}
 
-	// wait for the shimCtx to be done, which means the VM has exited and we're ready
-	// to shutdown
-	<-s.shimCtx.Done()
-	if shutdownCtx.Err() == context.DeadlineExceeded {
-		return status.Error(codes.DeadlineExceeded,
-			"timed out waiting for VM shutdown, VMM was sent SIGKILL")
-	}
+		s.logger.Debug("SIGTERM")
+		err = s.jailer.Stop()
+		if err != nil {
+			s.logger.WithError(err).Error("failed to stop the jailer")
+		} else {
+			time.Sleep(defaultSigtermTimeout)
+		}
 
-	if s.vmExitErr != nil {
-		return status.Error(codes.Internal,
-			fmt.Sprintf("VMM exit errors: %v", s.vmExitErr))
-	}
+		s.logger.Debug("SIGKILL")
```

Comment:
This is a little odd in terms of logging. Users are going to see `SIGTERM`, then always followed by `SIGKILL`. While this will `kill` runc, it really isn't a `SIGKILL`, but a delete of a container. We may want to actually send a `SIGKILL` then follow up with the `Close` function, if `SIGTERM` failed.

Reply:
I'm going to remove SIGTERM and SIGKILL from service.go since it is hidden under jailers mostly.

Reply:
Done.